lunabot_behavior/ascent.py

```diff
@@ -0,0 +1,59 @@
+import rospy
+
+from lunabot_msgs.msg import RobotEffort, RobotSensors
+
+import time
+
+EPSILON = 0.01 #TODO adjust as needed
+
+
+'''
+This is a transition state used to raise the linear actuators to the maximum height.
+'''
+class Ascent:
+
+	def sensors_callback(self, msg: RobotSensors):
+		self.robot_sensors = msg
+	
+	def __init__(self, effort_publisher: rospy.Publisher = None):
+		"""
+		If passed a publisher, then it is assumed a node is already running, and the publisher is shared.
+		Else, initialize this node to run on its own.
+		"""
+
+		if effort_publisher is None:
+			self.effort_publisher = rospy.Publisher("/effort", RobotEffort, queue_size=1, latch=True)
```

Comment:
What is the point of this logic? Why not always make the publisher here?

Reply:
the original point was to prevent us from making 2 nodes (b/c that doesn't work)
We could get rid of it? Not sure if it's bad practice to always be making and destroying publishers like that

Reply:
Oh yes I remember now.

I'm personally not a huge fan of this method, I'd prefer if the constructor for Ascent was only called once and functions within Ascent got called multiple times. within `behavior.py`, not sure how much that affects the rest of your implementation though.